code/game/objects/items/holy_weapons.dm

```diff
@@ -168,6 +168,7 @@
 	icon_state = "witchhunterhat"
 	item_state = "witchhunterhat"
 	flags_cover = HEADCOVERSEYES
+	flags_inv = HIDEEYES|HIDEHAIR
```

Comment:
this... isn't even what you wrote in the changelog. it still has the HIDEHAIR flag. Did you mean only HIDEEYES?